event hubs migration guide.md

```diff
@@ -0,0 +1,81 @@
+
+`EventHubClient` has been split into two clients - `EventHubProducerClient` (for sending
+messages) and `EventHubConsumerClient` (for subscribing to messages).
+
+| In v2                                       | Equivalent in v5                                                 | Sample |
+|---------------------------------------------|------------------------------------------------------------------|--------|
+| `EventHubClient.createFromConnectionString` | `new EventHubProducerClient()` or `new EventHubConsumerClient()` | [All](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/)
+| `EventHubClient.receive`                    | `EventHubConsumerClient.subscribe`                               | [receiveEvents](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/receiveEvents.ts) |
+| `EventHubClient.receiveBatch`               | Removed in favor of `EventHubConsumerClient.subscribe`           | [receiveEvents](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/receiveEvents.ts) |
+| `EventHubClient.createSender`               | `new EventHubProducerClient`                          | [sendEvents](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/sendEvents.ts) |
+| `EventHubClient.send`                       | `EventHubConsumerClient.sendBatch`                               | [sendEvents](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/sendEvents.ts) |
+
+### Migrating code from `EventHubClient` to `EventHubConsumerClient`
+
+In V2, event handlers were passed as positional arguments to `receive`.
+
+In V5, event handlers are passed as part of a `SubscriptionEventHandlers` shaped object.
+
+For example, this code which receives from a partition in V2:
+
+```typescript
+const rcvHandler = client.receive(partitionId, onMessageHandler, onErrorHandler, {
+  eventPosition: EventPosition.fromStart(),
+  consumerGroup: "$Default"
+});
+```
+
+Becomes this in V5:
+
+```typescript
+const eventHubConsumerClient = new EventHubConsumerClient("$Default", connectionString);
+
+const subscription = eventHubConsumerClient.subscribe(
+  partitionId, {
+    processInitialize: (initContext) => {
+      initContext.setStartingPosition(EventPosition.fromStart());
+    },
+    processEvents: onMessageHandler,
+    processError: onErrorHandler
+});
+  
+await subscription.close();
+```
+
+See [`receiveEvents.ts`](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/samples/receiveEvents.ts) 
+for a sample program demonstrating this.
+
+### Migrating code from `EventHubClient` to `EventHubProducerClient`
+
+In V2, there were multiple options on how to send data.
+
+In V5, this has been consolidated into a more efficient `sendBatch` method.
+
+So in V2:
+```typescript
+console.log(`Sending event: ${eventData.body}`);
+await client.send(eventData, partitionId);
```

Comment:
Since we want to highlight why sending with batches is better, should we consider the code snippet to reflect a function that takes an array of events? This is close to what we want to https://github.com/Azure/azure-sdk-for-js/issues/6254

The old sample would then be

```javascript
async function sendMyEvents(events) {
      await client.send(events); // Will fail if the total size of events exceed the max size supported by the library.
}
```

The newer one would be
```javascript
async function sendMyEvents(events) {
     const batch = await eventHubProducerClient.createBatch();
     let leftOverEvents = [];
     for (let i = 0; i < events.length; i++) {
        if (!batch.tryAdd(events[i]) {
               if (i === 0) {
                   // throw error here as we cannot batch the single event, too big
              }
               leftOverEvents = events.slice(i);
               break;
        }
     }
     await client.sendBatch(batch);
     await sendMyEvents(leftOverEvents);
}
```